From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Orson Zhai <orson.zhai@unisoc.com>
Date: Fri, 13 Mar 2020 22:51:10 +0800
Subject: ANDROID: gki_defconfig: Enable more USB_CONFIGFS gadgets

More USB gadgets are needed in Unisoc product like modules under
CONFIG_USB_CONFIGFS_SERIAL,
CONFIG_USB_CONFIGFS_RNDIS and
CONFIG_USB_CONFIGFS_MASS_STORAGE.

Bug: 151405649
Change-Id: If81e1d50f7b440f902e941d2450a551d0452314f
Signed-off-by: Orson Zhai <orson.zhai@unisoc.com>
[add x86 config chunk too - gregkh]
Signed-off-by: Greg Kroah-Hartman <gregkh@google.com>
---
 arch/arm64/configs/gki_defconfig | 3 +++
 arch/x86/configs/gki_defconfig   | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -335,6 +335,9 @@ CONFIG_USB_OTG=y
 CONFIG_USB_GADGET=y
 CONFIG_USB_CONFIGFS=y
 CONFIG_USB_CONFIGFS_UEVENT=y
+CONFIG_USB_CONFIGFS_SERIAL=y
+CONFIG_USB_CONFIGFS_RNDIS=y
+CONFIG_USB_CONFIGFS_MASS_STORAGE=y
 CONFIG_USB_CONFIGFS_F_FS=y
 CONFIG_USB_CONFIGFS_F_ACC=y
 CONFIG_USB_CONFIGFS_F_AUDIO_SRC=y
diff --git a/arch/x86/configs/gki_defconfig b/arch/x86/configs/gki_defconfig
--- a/arch/x86/configs/gki_defconfig
+++ b/arch/x86/configs/gki_defconfig
@@ -293,6 +293,9 @@ CONFIG_USB=y
 CONFIG_USB_GADGET=y
 CONFIG_USB_CONFIGFS=y
 CONFIG_USB_CONFIGFS_UEVENT=y
+CONFIG_USB_CONFIGFS_SERIAL=y
+CONFIG_USB_CONFIGFS_RNDIS=y
+CONFIG_USB_CONFIGFS_MASS_STORAGE=y
 CONFIG_USB_CONFIGFS_F_FS=y
 CONFIG_USB_CONFIGFS_F_ACC=y
 CONFIG_USB_CONFIGFS_F_AUDIO_SRC=y
